node/silkworm/db/mdbx.hpp

```diff
@@ -54,33 +56,116 @@ struct EnvConfig {
-
-// Executes a function on each record up to cursor.eof() or up to max_count (whichever the first).
-// To start from the beginning, point the cursor to the first entry by calling to_first().
-size_t for_count(::mdbx::cursor& cursor, const WalkFunc& func, size_t max_count);
+//! \brief Defines the direction of cursor while looping by cursor_for_each or cursor_for_count
+enum class CursorMoveDirection { Forward, Reverse };
+
+//! \brief Executes a function on each record reachable by the provided cursor
+//! \param [in] cursor : A reference to a cursor opened on a map
+//! \param [in] func : A pointer to a std::function with the code to execute on records. Note the return value of the
+//! function may stop the loop
+//! \param [in] direction : Whether the cursor should navigate records forward (default) or backwards
+//! \return The overall number of processed records
+//! \remarks If the provided cursor is *not* positioned on any record it will be moved to either the beginning or the
+//! end of the table on behalf of the move criteria
+size_t cursor_for_each(::mdbx::cursor& cursor, const WalkFunc& func,
+                       const CursorMoveDirection direction = CursorMoveDirection::Forward);
+
+//! \brief Executes a function on each record reachable by the provided cursor up to a max number of iterations
+//! \param [in] cursor : A reference to a cursor opened on a map
+//! \param [in] func : A pointer to a std::function with the code to execute on records. Note the return value of the
+//! function may stop the loop
+//! \param [in] max_count : Max number of iterations
+//! \param [in] direction : Whether the cursor should navigate records forward (default) or backwards
+//! \return The overall number of processed records. Should it not match the value of max_count it means the cursor has
+//! reached either the end or the beginning of table earlier
+//! \remarks If the provided cursor is *not* positioned on any record it will be moved to either the beginning or the
+//! end of the table on behalf of the move criteria
+size_t cursor_for_count(::mdbx::cursor& cursor, const WalkFunc& func, size_t max_count,
+                        const CursorMoveDirection direction = CursorMoveDirection::Forward);
+
+//! \brief Erases map records by cursor until any record is found
+//! \param [in] cursor : A reference to a cursor opened on a map
+//! \param [in] direction : Whether the cursor should navigate records forward (default) or backwards
+//! \return The overall number of erased records
+//! \remarks If the provided cursor is *not* positioned on any record it will be moved to either the beginning or the
+//! end of the table on behalf of the move criteria.
+//! \warning Might nuke all your table records if not used properly
+size_t cursor_erase(::mdbx::cursor& cursor, const CursorMoveDirection direction = CursorMoveDirection::Forward);
+
+//! \brief Erases map records by cursor until any record is found
```

Comment:
This sentence sounds unclear to me. To my mind the old description was clearer:
```
 * Truncate a table by starting from a certain key, either up or down.
 * If reverse is set to false, then every key either higher or equal than starting key will be erased.
 * If reverse is set to true, then every key either lower(not equal) than starting key will be erased.
 ```
Now it's not specified whether the entry with `set_key` is erased or not.

Reply:
Ah, OK, my bad, didn't notice that. However, "Erases map records by cursor until any record is found" sounds a bit weird to me, but it's a small thing.